search-solr-importer/data_import_handler.py

```diff
@@ -80,6 +91,50 @@ def collect_lear_data():
     return cur
 
 
+def is_good_standing(item_dict: dict, source: str) -> bool:  # pylint: disable=too-many-return-statements;
+    """Return the good standing value of the business."""
+    if item_dict['state'] != 'ACTIVE':
+        # Good standing is irrelevant to non-active businesses
+        return None
+    if source == 'LEAR':
+        if item_dict['legal_type'] in ['SP', 'GP']:
+            # Good standing is irrelevant to FIRMS
+            return None
+        if item_dict.get('restoration_expiry_date'):
+            # A business in limited restoration is not in good standing according to LEAR
+            return False
+        # rule directly from LEAR code
+        last_file_date = item_dict['last_ar_date'] or item_dict['founding_date']
+        return last_file_date + datedelta(years=1, months=2, days=1) > datetime.now(tz=timezone.utc)
+    # source == COLIN
+    if item_dict['corp_class'] in ['BC'] or item_dict['legal_type'] in ['LLC', 'LIC', 'A', 'B']:
+        if item_dict.get('state_typ_cd') in ['D1A', 'D1F', 'D1T', 'D2A', 'D2F', 'D2T']:
+            # Dissolution state is not in good standing
+            #   - updates into this state occur irregularly via batch job
+            #   - updates out of this state occur immediately when filing is processed
+            #     (can rely on this for a business being NOT in good standing only)
+            return False
+        if item_dict.get('xpro_jurisdiction') in ['AB', 'MB', 'SK']:
```

Comment:
these are the NWPTA provinces according to business, David R and the jurisdiction_type nwpta column in PROD